### DISCUSSION TITLE
LVGL backgorund fade looks unnormal

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20586] [Q&A]
is that an indicator for reversed lines in `display.ini`?
![WhatsApp Image 2024-01-24 at 23 12 32](https://github.com/arendst/Tasmota/assets/16272493/47309d10-d6a8-4a18-817c-96425c7d8635)

**Comment by sfromis** [comment]
Since you give zero details about your setup, maybe it could be an indication that your gpios for colors are off. Maybe try the gradient test I posted earlier, which should work well to check the gradients for both red, green and blue, with any "odd" striping (outside of the steps of the max available nuances) indicative of a setup mismatch.

**Reply by ChrisVonFrosta** [reply]
sorry. 
```
:H,EK9716_sunton,800,480,16,RGB,41,40,39,42,2,8,16,1,14,21,47,48,45,15,7,6,5,4,9,46,3,24
:S,2,1,1,0,40,20
:V,0,210,30,16,0,22,13,10,1
:0,00
:1,01
:2,02
:3,03
:TI1,5d,*,*,-1,38
#
```

```
... 
def set_default_layout()
        #- Background with a gradient from black #000000 (bottom) to dark blue #0000A0 (top) -#
        self.obj.set_style_bg_color(lv.color(0x040080), lv.PART_MAIN | lv.STATE_DEFAULT)
        self.obj.set_style_bg_grad_color(lv.color(0x000000), lv.PART_MAIN | lv.STATE_DEFAULT)
        self.obj.set_style_bg_grad_dir(lv.GRAD_DIR_VER, lv.PART_MAIN | lv.STATE_DEFAULT)
...
```

**Reply by ChrisVonFrosta** [reply]
where can i find this gradient test?

**Reply by sfromis** [reply]
https://github.com/arendst/Tasmota/discussions/20560#discussioncomment-8212834

(It also works even if you're not usually using HASPmota :wink: )

Since your blue gradient only appears to be from 80 to 00, you only have half the range, which I'd still expect to be 16 steps instead of your fewer stripes.

Not sure if your experiment using clock speed 24 MHz instead of the usual 14 might make a difference, but my fully working gradients are seen with 14 MHz, 32 stripes for red/blue and 64 for green (the latter reasonably smooth).

**Reply by s-hadinger** [reply]
You can add dithering to avoid those visible layers:
```berry
def set_default_layout()
        #- Background with a gradient from black #000000 (bottom) to dark blue #0000A0 (top) -#
        self.obj.set_style_bg_color(lv.color(0x040080), lv.PART_MAIN | lv.STATE_DEFAULT)
        self.obj.set_style_bg_grad_color(lv.color(0x000000), lv.PART_MAIN | lv.STATE_DEFAULT)
        self.obj.set_style_bg_grad_dir(lv.GRAD_DIR_VER, lv.PART_MAIN | lv.STATE_DEFAULT)

        self.obj.set_style_bg_dither_mode(lv.DITHER_ORDERED, lv.PART_MAIN | lv.STATE_DEFAULT)
```

**Reply by ChrisVonFrosta** [reply]
oh nice. thank you for pointing to. yesterday i flashed a first test in C. i will try as soon as i reflashed tasmota or i got running this [display ](https://github.com/arendst/Tasmota/discussions/20558#discussioncomment-8250619) ;)
